### PR TITLE
Feature/dhclient configs

### PR DESCRIPTION
**Pull request by metelik**
Adding support for DHCP protocol timings configuration

### CODE REVIEW

File: lib/nerves_network/dhcp_manager.ex [hunk 1]
```diff
@@ -146,8 +147,9 @@ defmodule Nerves.Network.DHCPManager do
 
   ## covers STOP, RELEASE, FAIL and EXPIRE in Nerves.Network.Dhclientv4
   defp consume(:dhcp, :ifdown, state), do: consume(:dhcp, {:ifdown, :no_info}, state)
+
   defp consume(:dhcp, {:ifdown, info}, state) do
-    Logger.debug fn -> "#{__MODULE__}: consume (context = :dhcp) :ifup info: #{inspect info}" end
+    Logger.debug fn -> "#{__MODULE__}: consume (context = :dhcp) :ifdown info: #{inspect info}" end
```
Comment (michaelkschmidt) [inline lib/nerves_network/dhcp_manager.ex]: Minor nit-pick:  We don't need to wrap in `fn -> end` anymore as that was fixed in the latest Elixir.  Also __MODULE__ is printed out by the Logger module format string now too

**Review by michaelkschmidt — APPROVED** [review]
Looks good

File: lib/nerves_network/dhcp_manager.ex (hunk 1 above)
Comment (metelik) [reply lib/nerves_network/dhcp_manager.ex]: Yup this is the residue from the previous implementation

Comment (metelik) [reply lib/nerves_network/dhcp_manager.ex]: went through the unit so it now neither uses the fn -> nor \#{__MODULE__}

**Pull request merged**